docs/source/performance.rst

```diff
@@ -0,0 +1,174 @@
+See :doc:`tips` for steps to run your model with iDeep.
+
+You can also improve performance by building NumPy linked to `Intel MKL <https://software.intel.com/en-us/mkl>`__.
+See `Numpy/Scipy with Intel® MKL and Intel® Compilers <https://software.intel.com/en-us/articles/numpyscipy-with-intel-mkl>`__ for the detailed instructions.
+
+.. note::
+
+    If you installed `numpy <https://anaconda.org/anaconda/numpy>`__ package using Anaconda, you may already have MKL-linked NumPy.
+    Check the output of ``numpy.show_config()`` to see what linear algebra library is linked.
+
+.. note::
+
+    Use of iDeep and MKL-linked NumPy are orthogonal.
+    You can use both of them at once to maximize the performance.
+
+Migrate Data Preprocessing Code from NumPy to CuPy
+--------------------------------------------------
+
+If you are preprocessing your dataset or running data augmentation using NumPy, you may be able to use CuPy as a substitution to improve performance.
+
+.. note::
+
+    It is **not always** efficient to use CuPy instead of NumPy, especially when the computation is not very heavy, or it cannot be done in batch.
+
+Avoid Data Transfer
+-------------------
+
+If you are using GPU, be aware of data transfer between CPU and GPU.
+For example, ``print``\ing :class:`chainer.Variable` on GPU (e.g., for debugging) will cause memory transfer from GPU to CPU, which will incur synchronization overhead.
+
+You can use `NVIDIA Visual Profiler <https://docs.nvidia.com/cuda/profiler-users-guide/>`__ to diagnose this kind of issue.
+
+Optimize cuDNN Convolution
+--------------------------
+
+Workspace Size
+~~~~~~~~~~~~~~
+
+Some convolution algorithms in cuDNN use additional GPU memory as a temporary buffer.
+This is called "workspace," and users can adjust the upper limit of its size.
+By increasing the limit of workspace size, cuDNN may be able to use better (i.e., memory consuming but faster) algorithm.
+
+The default size (in bytes) is:
+
+.. doctest::
+
+    >>> chainer.backends.cuda.get_max_workspace_size()
+    8388608
+
+and can be adjusted using :func:`chainer.backends.cuda.set_max_workspace_size`.
+
+Maximum required workspace size may vary depending on various conditions such as GPU hardware and batch size of inputs.
+
+Auto-Tuner
+~~~~~~~~~~
+
+Some convolution algorithms in cuDNN support auto-tuner feature that finds the fastest convolution algorithm for given inputs.
+You can turn on this feature by setting ``autotune`` configuration to ``True``.
+
+See :doc:`reference/configuration` for detailed descriptions.
+
+.. note::
+
+    Auto-tuner tries to find the best algorithm for every first observation of the input shape combination.
+    Therefore, the first batch will become slower when auto-tuner is enabled.
+    The result of auto-tuner is cached so that it can be reused for data with the same input shape combination.
```

Comment:
Will the cached result of the auto-tuner be reused if the model is later run with ``autotuned`` set to ``False``? This is not clear to me from the description. Or should the user then set it to ``False`` for later runs?